dipy/reconst/ivim.py

```diff
@@ -153,20 +166,18 @@ def nlls_fit(data, gtab, guess_params):
 
     Returns
     -------
-    nlls_params: the S0, f, D_star, D value for each voxel.
+    ivim_params: the S0, f, D_star, D value for each voxel.
 
     """
     flat_data = data.reshape((-1, data.shape[-1]))
     # Flatten for the iteration over voxels:
     bvals = gtab.bvals
     ivim_params = np.empty((flat_data.shape[0], 4))
     for vox in range(flat_data.shape[0]):
-        popt, pcov = curve_fit(ivim_function, bvals, flat_data[vox],
-                               guess_params[vox])
-        ivim_params[vox, :4] = popt
+        res = leastsq(_ivim_error,
```

Comment:
The function (`ivim_function`) needs the first argument to be what it's trying to solve. So in this case it would be a vector (numpy array) of S0, f, D, D*. But you don't send that argument in with args, so your function is `ivim_function(x,bvals)` and your call is `minimize(ivim_function,x0[vox],args=(bvals,))` I think that's right at least without testing.